### PR TITLE
Contract Interaction page overhaul

### LINKED ISSUE #3127: Unexpected behavior when using the parameters feature in the "Interact with Contracts" page
### Description of Issue

Contract address is not resolved when passing parameters in the url of the "Interact with Contracts" page. Despite the "Select Existing Contract" is well selected, the "Contract Address" field stay empty and a red exclamation mark is displayed.

image

Please note that when the contract is selected "manually", the address is well resolved.

image

### What exactly were you doing when you encountered this issue?

Just follow this link:
https://beta.mycrypto.com/interact-with-contracts?network=Ethereum&address=0x90bb0b538253d279642439d5f54f336414ec06f7&function=issuers&input=0:49

### How were you accessing your wallet?

no wallet needed

### Description of Your Machine

- Browser: chrome
- Browser version:  80.0.3987.149 (Official Build) (64-bit)
- Operating system: Windows 10
- Device: Microsoft Surface Pro
- MyCrypto.com vs Desktop Application? MyCrypto.com
- MyCrypto Version Number: beta

### PR DESCRIPTION
**Pull request by FrederikBolding**
Fixes #3127 

Furthermore cleans up error messaging and ENS for contract interaction page.
- Abstracted out `GeneralLookupField` to prevent creating two identical implementations of the same thing.
- Replaced contract selection UI with new `ContractLookupField`

### CODE REVIEW

**Comment by FrederikBolding** [comment]
[ch6064]

**Comment by FrederikBolding** [comment]
Waiting for design feedback on which type of inline message to show for ENS resolved address.

**Comment by blurpesec** [comment]
@FrederikBolding after clicking on the `MyCrypto Membership 1-Month` contract and selecting the write function `expireAndRefundFor` I get a RSOD and reroute to home page with err: 
image

**Comment by FrederikBolding** [comment]
@blurpesec I can't reproduce this, is this due to localstorage thing? Looks like the crash with the removed base assets.

**Comment by blurpesec** [comment]
> @blurpesec I can't reproduce this, is this due to a localstorage thing? Looks like the crash with the removed base assets.

looks like it. i made a pr to stop this bug in #3294

**Comment by blurpesec** [comment]
The Contract Selection dropdown seems to be broken now - cleared cache and tested again. 
image

File: src/components/GeneralLookupField.tsx [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+      );
+      if (UnstoppableResolution.isResolutionError(err)) {
+        setResolutionError(err);
+      } else throw err;
+    } finally {
+      setIsResolvingDomain(false);
+    }
+  };
+
+  const GeneralDropdownFieldCallback = useCallback(() => {
```
Comment (celowallet) [inline src/components/GeneralLookupField.tsx:156]: Can you add some tests for this field to make sure name resolution and inputs are correctly validated and displaying correct error and loading states?

File: src/features/InteractWithContracts/stateFactory.tsx [hunk 2]
```diff
@@ -74,9 +74,9 @@ const InteractWithContractsFactory: TUseStateReducerFactory<InteractWithContract
     const contractIds = state.network.contracts;
     const networkContracts = getContractsByIds(contractIds);
 
-    const customContractOption = Object.assign({}, customContract, { networkId: state.network.id });
+    //const customContractOption = Object.assign({}, customContract, { networkId: state.network.id });
```
Comment (celowallet) [inline src/features/InteractWithContracts/stateFactory.tsx:77]: if this isn't needed, can't we delete?

File: src/translations/lang/en.json [hunk 3]
```diff
@@ -150,7 +150,8 @@
     "CONTRACT_JSON": "ABI / JSON Interface ",
     "CONTRACT_NAME": "Contract Name",
     "CONTRACT_NAME_PLACEHOLDER": "Custom contract name",
-    "CONTRACT_TITLE": "Contract Address ",
+    "CONTRACT_SELECTION_PLACEHOLDER": "Enter an Address or choose a Contract",
```
Comment (celowallet) [inline src/translations/lang/en.json:153]: "Enter an Address or choose a Contract" => "Enter an address or choose a contract"

File: src/components/GeneralLookupField.tsx [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+  onBlur?(): void;
+  setIsResolvingDomain(isResolving: boolean): void;
+  handleEthAddress(inputString: string): IReceiverAddress;
+  handleENSName(resolvedAddress: string, inputString: string): IReceiverAddress;
+  onSelect?(option: IReceiverAddress): void;
+  onChange?(input: string): void;
+  onLoad?(setFieldValue: (field: string, value: any, shouldValidate?: boolean) => void): void;
+}
+
+const GeneralLookupField = ({
```
Comment (blurpesec) [inline src/components/GeneralLookupField.tsx:32]: Can you add some tests for this field to make sure name resolution and inputs are correctly validated and displaying correct error and loading states?

File: src/features/InteractWithContracts/stateFactory.tsx (hunk 2 above)
Comment (blurpesec) [inline src/features/InteractWithContracts/stateFactory.tsx:77]: if this isn't needed, can't we delete?

File: src/translations/lang/en.json (hunk 3 above)
Comment (blurpesec) [inline src/translations/lang/en.json:153]: "Enter an Address or choose a Contract" => "Enter an address or choose a contract"

**Pull request merged**